### PR TITLE
1037 - Cache clear mechanism

### PR DESCRIPTION
**Pull request by brownbearsoftware**
This adds a `DELETE /cache/{cache name}/clear` endpoint that can only be accessed locally which clears the named cache.

### CODE REVIEW

**Comment by brownbearsoftware** [comment]
[Add mechanism for clearing API caches without releasing a new build](https://trello.com/c/GjkGZYO2/1037-add-mechanism-for-clearing-api-caches-without-releasing-a-new-build)

File: src/main/resources/static/api.yml [hunk 1]
```diff
@@ -2628,6 +2628,24 @@ paths:
           $ref: '#/components/responses/403Response'
         500:
           $ref: '#/components/responses/500Response'
+  /cache/{cacheName}/clear:
```
Comment (pezholio) [inline src/main/resources/static/api.yml:2631]: With my REST hat on, we should just call this `/cache/{cacheName}`, as `cacheName` is the thing we're `DELETE`ing

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/CacheClearTest.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+      // Call our endpoint again
+      webTestClient.get()
+        .uri("/premises/$premisesId/staff")
+        .header("Authorization", "Bearer $jwt")
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      // Validate upstream request was made again, proving the cache was cleared
```
Comment (pezholio) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/CacheClearTest.kt:51]: Loving these comments - makes it so much easier to follow! Might be worth following the convention we use in the UI and use Cucumber-like syntax, ie:

- Given I call the qCode endpoint twice
- Then the upstream request should only have been made once
- When I clear the qCode cache
- And I call the endpoint again
- Then the upstream request should be made again

**Review by pezholio — APPROVED** [review]
LGTM - Just thinking about how to make those comments even more useful. Might be worth copying for future integration tests too. Often there's a lot going on, so cucumber-like syntax makes it easy to follow 👍

File: src/main/resources/static/api.yml (hunk 1 above)
Comment (brownbearsoftware) [reply src/main/resources/static/api.yml]: Good point, have switched it

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/CacheClearTest.kt (hunk 2 above)
Comment (brownbearsoftware) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/CacheClearTest.kt]: Done for that one and the other test too

**Review by danlivings-dxw — APPROVED** [review]
LGTM

**Pull request merged**